microceph-orch/src/microceph/client/service.py

```diff
@@ -0,0 +1,218 @@
+# SPDX-FileCopyrightText: 2023 - Canonical Ltd
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+from abc import ABC
+
+from requests.exceptions import ConnectionError, HTTPError
+from requests.sessions import Session
+
+LOG = logging.getLogger(__name__)
+
+
+class RemoteException(Exception):
+    """An Exception raised when interacting with the remote microclusterd service."""
+
+    pass
+
+
+class ClusterAlreadyBootstrappedException(RemoteException):
+    """Raised when cluster service is already bootstrapped."""
+
+    pass
+
+
+class ClusterServiceUnavailableException(RemoteException):
+    """Raised when cluster service is not yet bootstrapped."""
+
+    pass
+
+
+class ConfigItemNotFoundException(RemoteException):
+    """Raise when ConfigItem cannot be found on the remote."""
+
+    pass
+
+
+class NodeAlreadyExistsException(RemoteException):
+    """Raised when the node already exists."""
+
+    pass
+
+
+class NodeNotExistInClusterException(RemoteException):
+    """Raised when the node does not exist in cluster."""
+
+    pass
+
+
+class NodeJoinException(RemoteException):
+    """Raised when the node not able to join cluster."""
+
+    pass
+
+
+class LastNodeRemovalFromClusterException(RemoteException):
+    """Raised when token is already generated for the node."""
+
+    pass
+
+
+class TokenAlreadyGeneratedException(RemoteException):
+    """Raised when token is already generated for the node."""
+
+    pass
+
+
+class TokenNotFoundException(RemoteException):
+    """Raised when token is not found for the node."""
+
+    pass
+
+
+class URLNotFoundException(RemoteException):
+    """Raise when URL is not found.
+
+    Happens when the URL is not found in the remote or
+    microcluster is not bootstrapped.
+    """
+
+    pass
+
+
+class BaseService(ABC):
+    """BaseService is the base service class for sunbeam clusterd services."""
+
+    def __init__(
+        self,
+        session: Session,
+        endpoint: str,
+        certs=None,
+        timeout: int | float | None = None,
+    ):
+        """Creates a new BaseService for the sunbeam clusterd API.
```

Comment:
In initial draft, this client was taken from the sunbeam code base, but the references to sunbeam clusterd will feel out of place to future authors. I propose we remove the references and potentially get rid of some of the token/node management pieces. They can always be added back later if / when it is necessary.

Reply:
ack, done. Dropped unrelated references and token methods.